From fd802c0a074f81d2c5ac2e604408923f7e2adc3e Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Wed, 11 Jan 2012 04:42:44 -0500
Subject: [PATCH] Deal properly with scheduled IERS transitions in the
 future...

...as well as past ones. All regression tests pass.
---
 leapsecond.py |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/leapsecond.py b/leapsecond.py
index 4848818..2059f6c 100755
--- a/leapsecond.py
+++ b/leapsecond.py
@@ -138,10 +138,13 @@ def fetch_leapsecs(filename):
     return leapsecs
 
 def make_leapsecond_include(infile):
-    leapsecs = fetch_leapsecs(infile)
-    leapsecs.append(time.time())          # Add sentinel
+    leapjumps = fetch_leapsecs(infile)
     year = time.strftime("%Y", time.localtime(time.time()))
-    return ("#define CENTURY_BASE\t%s00\n" % year[:2]) + ("#define LEAPSECOND_NOW\t%d\n" % (len(leapsecs)-2))
+    leapsecs = 0
+    for leapjump in leapjumps:
+        if leapjump < time.time():
+            leapsecs += 1
+    return ("#define CENTURY_BASE\t%s00\n" % year[:2]) + ("#define LEAPSECOND_NOW\t%d\n" % (leapsecs-1))
 
 def leastsquares(tuples):
     "Generate coefficients for a least-squares fit to the specified data."
-- 
1.7.7.1

